mixer/pkg/api/grpcServer.go

```diff
@@ -273,18 +231,17 @@ func (s *grpcServer) Report(legacyCtx legacyContext.Context, req *mixerpb.Report
 	}
 
 	protoBag := attribute.NewProtoBag(&req.Attributes[0], s.globalDict, s.globalWordList)
-	requestBag := attribute.GetMutableBag(protoBag)
-	compatReqBag := &compatBag{requestBag}
-	mutableBag := attribute.GetMutableBag(requestBag)
-	preprocResponseBag := attribute.GetMutableBag(nil)
+	accumBag := attribute.GetMutableBag(protoBag)
```

Comment:
A comment about why each bag is used would be great. It is hard to parse and grok what the function is doing, even more so with the loop and various uses of the bags. (Especially since the bag hierarchy is preserved across the iterations, but the individual bags get updated.)